posts/transform-data-in-your-warehouse-using-dbt-and-airflow.md

```diff
@@ -0,0 +1,138 @@
+    wr.redshift.to_sql(
+        df=df,
+        table="quote",
+        schema="public",
+        con=con
+    )
+    con.close()
+
+
+with DAG("bitcoin-price", schedule_interval="*/5 * * * *", start_date=datetime(2022, 4, 5), catchup=False) as dag:
+    save_quote_task = PythonOperator(task_id="save-quote",
+                                     python_callable=save_quote)
+
+    dbt_task = DbtRunOperator(task_id="dbt",
+                              dbt_bin="/usr/local/airflow/.local/bin/dbt",
+                              profiles_dir="/usr/local/airflow/dags/blog_dbt_airflow/",
+                              dir="/usr/local/airflow/dags/blog_dbt_airflow/",
+                              models="quote")
+
+    save_quote_task >> dbt_task
+```
+
+Running DBT directly on Airflow is possible using the operators from [this GitHub project](https://github.com/gocardless/airflow-dbt), in this case, we will simply use the `DbtRunOperator` and specify where dbt is installed. The CoinMarketCap API Key is stored as an Airflow variable to avoid having it hard-coded.
+
+# DBT Model
+
+The DBT model is pretty simple since the goal of this blog post is to show how DBT can run directly on Airflow (instead of an ECS task for example). All the DBT code will be stored directly in the S3 Bucket so the Airflow DAG can easily access it.
+
+Here is the incremental DBT model that will simply add new rows based on the column `inserted_at`:
+
+```sql
+{{
+    config(
+        materialized='incremental'
+    )
+}}
+
+SELECT * FROM public.quote
+
+{% if is_incremental() %}
+  WHERE inserted_at > (SELECT max(inserted_at) FROM {{ this }})
+{% endif %}
+```
+
+The target schema for this model is `dbt`, for more details, you can take a look at the file `profiles.yml`. Few properties from the file `dbt_project.yml` were also changed because of read-only file system (`packages-install-path`, `log-path` and `target-path`).
+
+# Deployment using AWS CDK
+
+The whole stack can be deployed with [CDK](https://aws.amazon.com/cdk/) using the command `cdk deploy`, it will generate a CloudFormation template and execute it.
+
+Stack's components:
+- Clean VPC
+- S3 Bucket prefilled with Airflow DAG and DBT code
+- IAM Role with mandatory permissions
+- Redshift cluster
+- Airflow
+
+Feel free to clone my [GitHub repository](https://github.com/Falydoor/blog-dbt-airflow) that has the stack already configured.
+
+Once the whole stack is deployed and the Airflow instance running, the DAG can be turned on and it will start collecting data and transforming it.
+
+You can see the DBT logs directly on the Airflow UI:
+
+![](https://raw.githubusercontent.com/Falydoor/blog-usa/blog-dbt-airflow/images/2022/04/dbt-airflow-logs.png)
+
+# Conclusion
+
+Creating a project that uses DBT and Airflow was relatively easy using the available AWS services. CDK really lets you focus on the architecture design instead of wasting time on trivial things like configurations, permissions and networking.
```

Comment:
```suggestion
Creating a project that uses dbt and Airflow was relatively easy using the available AWS services. CDK allows you to focus on the architecture design instead of wasting time on trivial things like configurations, permissions, and networking.
```